src/libraries/System.Net.Http/tests/FunctionalTests/HttpConnectionKeyTest.cs

```diff
@@ -23,7 +23,8 @@ public static IEnumerable<object[]> KeyComponents()
             yield return new object[] { "Http", "localhost", 80, "localhost-ssl", new Uri("http://localhost"), "domain1/userA", true };
         }
 
-        [Theory, MemberData(nameof(KeyComponents))]
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBuiltWithAggressiveTrimming))]
```

Comment:
It would be better to replace the insane expensive way to get the type using Linq by just calling `GetType`.

Reply:
There's also `dynamic` keyword below that wasn't obvious to me why it was added. I assume there's a reason and not just whoever wrote it doesn't know what they're doing.

Reply:
> I assume there's a reason and not just whoever wrote it doesn't know what they're doing.

I do not think there is a reason. This whole method was clearly written by somebody who did not know what they were doing.